internal/support/notifications/rest_notification.go

```diff
@@ -345,20 +345,16 @@ func notificationByEndHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	n, err := dbClient.GetNotificationsByEnd(end, limitNum)
+	op := notification.NewEndExecutor(dbClient, end, limitNum)
+	results, err := op.Execute()
 	if err != nil {
-		if err == db.ErrNotFound {
-			http.Error(w, "Notification not found", http.StatusNotFound)
-		} else {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
We can't stop returning 404 here since that's what the published v1.0.0 API does. In the Execute() implementation check to see if there are zero results. If so, return a db.ErrNotFound
```
if len(res) == 0 {
	return res, db.ErrNotFound
}
```
From there you can preserve the previous type checking and send the appropriate http code.